scripts/update_substrate_git_branch.sh

```diff
@@ -4,12 +4,15 @@ set -e
 
 branch="$1"
 
-paths=(bin/node/Cargo.toml bin/runtime/Cargo.toml finality-aleph/Cargo.toml
-    primitives/Cargo.toml pallet/Cargo.toml)
+# Find all `Cargo.toml` files outside any `target` directory.
+paths=$(find . -mindepth 2 -type f -name "Cargo.toml" -not -path "*/target/*")
 
 for path in ${paths[@]}; do
     echo upgrade $path
-    sed -e 's/\(substrate.git.*branch = "\)[^"]*".*\([,}]\)/\1'$branch'" \2/' < $path > x
+    # 1. Find and capture `paritytech/substrate.git", branch = "` substring. It will be available as `\1`.
+    # 2. Find and capture whatever is after closing `"` and before `,` or `}`. It will be available as `\2`.
+    # 3. Substitute new branch and concatenate it with `\1` and `\2`.
+    sed -e 's/\(paritytech\/substrate.git", branch = "\)[^"]*"\([^,}]*\)/\1'$branch'"\2/' < $path > x
```

Comment:
You can make it little bit less error prone by adding `\s` here and there (e.g. between branch and =).

Reply:
done